api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1725,17 +1725,27 @@ type AzurePlatformSpec struct {
 
 	// ResourceGroupName is the name of an existing resource group where all cloud resources created by the Hosted
 	// Cluster are to be placed.
+	// Resource group naming requirements can be found here: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.ResourceGroup.Name/.
 	// In ARO HCP, this will be the managed resource group where customer cloud resources will be created.
 	//
 	// Example: if your resource group ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>, your
 	//          ResourceGroupName is <resourceGroupName>.
 	//
 	// +kubebuilder:default:=default
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_()\-\.]{1,89}[a-zA-Z0-9_()\-]$`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
 	// +immutable
 	ResourceGroupName string `json:"resourceGroup"`
```

Comment:
Given that this is resource group name I would document that the Resource Group must belong to the specified subscription in the `SubscriptionID` attribute